python/tests/conftest.py

```diff
@@ -72,468 +38,163 @@
+def version_store_factory(lib_name, lmdb_storage):
+    return lmdb_storage.create_version_store_factory(lib_name)
 
 
 @pytest.fixture
-def azurite_azure_uri_incl_bucket(azurite_azure_uri, azure_client_and_create_container):
-    return azurite_azure_uri
-
-
-@pytest.fixture(scope="function")
-def arctic_library(request, arctic_client):
-    arctic_client.create_library("pytest_test_lib")
-
-    yield arctic_client["pytest_test_lib"]
-
-
-@pytest.fixture()
-def sym():
-    return "test" + datetime.utcnow().strftime("%Y-%m-%dT%H_%M_%S_%f")
-
-
-@pytest.fixture()
-def lib_name():
-    return f"local.test.{random.randint(0, 999)}_{datetime.utcnow().strftime('%Y-%m-%dT%H_%M_%S_%f')}"
+def s3_store_factory(lib_name, s3_storage):
+    return s3_storage.create_version_store_factory(lib_name)
 
 
 @pytest.fixture
-def arcticdb_test_s3_config(moto_s3_endpoint_and_credentials):
-    endpoint, port, bucket, aws_access_key, aws_secret_key = moto_s3_endpoint_and_credentials
-
-    def create(lib_name):
-        return create_test_s3_cfg(lib_name, aws_access_key, aws_secret_key, bucket, endpoint)
-
-    return create
+def real_s3_store_factory(lib_name, real_s3_storage):
+    return real_s3_storage.create_version_store_factory(lib_name)
 
 
 @pytest.fixture
-def arcticdb_test_azure_config(azurite_azure_test_connection_setting, azurite_azure_uri):
-    def create(lib_name):
-        (endpoint, container, credential_name, credential_key, ca_cert_path) = azurite_azure_test_connection_setting
-        return create_test_azure_cfg(
-            lib_name=lib_name,
-            credential_name=credential_name,
-            credential_key=credential_key,
-            container_name=container,
-            endpoint=azurite_azure_uri,
-            ca_cert_path=ca_cert_path,
-        )
-
-    return create
-
-
-def _version_store_factory_impl(
-    used, make_cfg, default_name, *, name: str = None, reuse_name=False, **kwargs
-) -> NativeVersionStore:
-    """Common logic behind all the factory fixtures"""
-    name = name or default_name
-    if name == "_unique_":
-        name = name + str(len(used))
-
-    assert (name not in used) or reuse_name, f"{name} is already in use"
-    cfg = make_cfg(name)
-    lib = cfg.env_by_id[Defaults.ENV].lib_by_path[name]
-    # Use symbol list by default (can still be overridden by kwargs)
-    lib.version.symbol_list = True
-    apply_lib_cfg(lib, kwargs)
-    out = ArcticMemoryConfig(cfg, Defaults.ENV)[name]
-    used[name] = out
-    return out
-
-
-@enum.unique
-class EncodingVersion(enum.IntEnum):
-    V1 = 0
-    V2 = 1
-
-
-@pytest.fixture(scope="session")
-def only_test_encoding_version_v1():
-    """Dummy fixture to reference at module/class level to reduce test cases"""
-
-
-def pytest_generate_tests(metafunc):
-    if "encoding_version" in metafunc.fixturenames:
-        if sys.platform == "win32":
-            metafunc.parametrize("encoding_version", [EncodingVersion.V2])
```

Comment:
Is dropping this going to slow down the Windows builds? Thinking about it though, the old logic is really weird, since we still have `DEFAULT_ENCODING_VERSION = EncodingVersion.V1` on all platforms

Reply:
It will be slower, but I felt not testing both encoding versions is wrong.

I was thinking about using `mem` storage for V1 testing on Windows, but once I implement the LMDB reuse logic, it should be roughly as fast as `mem`.